src/main/java/com/mongodb/jdbc/MongoSQLDatabaseMetaData.java

```diff
@@ -106,11 +129,126 @@ public ResultSet getTableTypes() throws SQLException {
         throw new SQLFeatureNotSupportedException("TODO");
     }
 
+    // Helper for creating BSON documents for the getTables method. Intended for use
+    // with the getTablesFromDB helper method which is shared between getTables and
+    // getTablePrivileges.
+    private BsonDocument toGetTablesDoc(String dbName, MongoListCollectionsResult res) {
+        return createBottomBson(
+                new BsonElement(TABLE_CAT, new BsonString(dbName)),
+                new BsonElement(TABLE_SCHEM, new BsonString("")),
+                new BsonElement(TABLE_NAME, new BsonString(res.name)),
+                new BsonElement(TABLE_TYPE, new BsonString(res.type)),
+                new BsonElement(REMARKS, new BsonNull()),
+                new BsonElement(TYPE_CAT, new BsonNull()),
+                new BsonElement(TYPE_SCHEM, new BsonNull()),
+                new BsonElement(TYPE_NAME, new BsonNull()),
+                new BsonElement(SELF_REFERENCING_COL_NAME, new BsonNull()),
+                new BsonElement(REF_GENERATION, new BsonNull()));
+    }
+
+    // Helper for creating BSON documents for the getTablePrivileges method. Intended
+    // for use with the getTablesFromDB helper method which is shared between getTables
+    // and getTablePrivileges.
+    private BsonDocument toGetTablePrivilegesDoc(String dbName, MongoListCollectionsResult res) {
+        return createBottomBson(
+                new BsonElement(TABLE_CAT, new BsonString(dbName)),
+                new BsonElement(TABLE_SCHEM, new BsonString("")),
+                new BsonElement(TABLE_NAME, new BsonString(res.name)),
+                new BsonElement(GRANTOR, new BsonNull()),
+                new BsonElement(GRANTEE, new BsonString("")),
```

Comment:
The JDBC javadocs say `GRANTOR` may be null but does not say the same for `GRANTEE`. The design doc said to make these `null` or `""`, so I went with `null` for the former since it is valid there and empty string for the latter since `null` is not valid there.